controller/monitor/node_monitor.go

```diff
@@ -250,15 +257,42 @@ func (m *NodeMonitor) getOrphanedReplicaDirectoryNames(node *longhorn.Node, disk
 		}
 	}
 
+	// Find out the new/missing orphaned directories by checking with orphan CRs
```

Comment:
Not sure why you move this logic to the node monitor... I think it's NodeController's duty to filter out the mismatching orphans...

Reply:
I though the previous checking is hard to read. Maybe we can move it back and group them into new and missing after node controller gets the new list.

Reply:
Another consideration is that if we separate new and missing dirs in node monitor, then we don't have to do `isVolumeMetaFileExist` for all dirs blindly in the node monitor every 30 seconds.

Reply:
As I mentioned, the monitor is responsible for detecting the disk status only regardless of how NodeController reconciles the status and handles the orphans.
After renaming the related fields or functions, putting these checks in `createOrphans` or `deleteOrphans` is not as messy as you are worried about.

And I was thinking if you are trying to avoid the unnecessary `isVolumeMetaFileExist` as well. 
IMO, the cost should not be expensive. Even if there are lots of missing orphans (extra CRs) here, the CRs will be removed soon and those unnecessary metafile checks are one-time operations.